CMakeLists.txt

```diff
@@ -27,6 +27,8 @@ cmake_minimum_required(VERSION 2.8.12 FATAL_ERROR)
 
 project(pocl)
 
+set(CMAKE_CXX_STANDARD 20) # probes the standards and sets the highest <=20
```

Comment:
The purpose of this is to allow `[[maybe_unused]]` for the compilers I have not explicitly implemented support in a compiler-specific way. `[[maybe_unused]]` is available in 17, but since we are raising, I feel like we should raise to the top.

>Shouldn't we actually set it to the lowest standard required to avoid accidental introduction of too new features to the code base?

I guess it should be the problem of the ones using ancient compilers. I mean if they found something that breaks the code from compilation in their compiler, it's they who should 
1. decide if it is feasible to them to just upgrade the compiler (in most cases just installing the latest gcc/clang is enough and it usually doesn't require hardware upgrades)
2. if it is infeasible, create the code for the older compiler and send the PR.

Reply:
Hi KOLANICH,

1. Is not feasible in many cases, often people are not admins or responsible of being the admin of their installations. I do not want force users to this type of troubles if not really necessary.
2. Unfortunately very few pocl users are actually ready to go through the steps required to contribute code and push PR (you are in a minority here!), but if they face problems they just don't use the code and stamp the project as sloppy and unportable in their minds. While we claim we support building with C++11 (whatever LLVM builds with), we should do our best to deliver that promise.

Adding [[maybe_unused]] support is not a major enough motivation to start passively breaking support for building pocl with C++11. If you encounter a compiler that has this attribute implemented, but doesn't have one of the explicit cases you define, that's a good PR, not the other way around.

Reply:
I have put that behind a switch disabled by default. By default it uses 11.

Reply:
Thanks!